Packs/Kafka/Integrations/KafkaV3/KafkaV3.py

```diff
@@ -0,0 +1,633 @@
+                 offset: str = None, consumer: bool = False) -> bool:
+    """Check that partitions exist in topic and that offset matches the available ones.
+
+    Args:
+        kafka: Initialized KafkaCommunicator object to preform actions with.
+        topic: The topic
+        partitions: list of partitions
+        offset: The offset
+        consumer: consumer: Whether to get topics using the consumer's configuration (True) or producer's (False).
+
+    Return True if everything is valid, raise relevant exception otherwise.
+    """
+    check_offset = False
+    topics = kafka.get_topics(consumer=consumer)
+    if topic not in topics.keys():
+        raise DemistoException(f"Did not find topic {topic} in kafka topics.")
+
+    if offset and str(offset).lower() not in SUPPORTED_GENERAL_OFFSETS:
+        if offset.isdigit():
+            offset = int(offset)  # type: ignore
+            check_offset = True
+        else:
+            raise DemistoException(f'Offset {offset} is not in supported format.')
+
+    if partitions:
+        topic_metadata = topics[topic]
+        available_partitions = topic_metadata.partitions.values()
+        available_partitions_ids = [available_partition.id for available_partition in available_partitions]
+        for partition in partitions:
+            if int(partition) not in available_partitions_ids:
+                raise DemistoException(f"Partition {partition} is not assigned to kafka topic {topic} available "
+                                       f"{available_partitions_ids}.")
+            if check_offset:
+                earliest_offset, oldest_offset = kafka.get_partition_offsets(topic=topic, partition=int(partition))
+                if offset < int(earliest_offset) or offset >= int(oldest_offset):  # type: ignore
+                    raise DemistoException(f'Offset {offset} for topic {topic} and partition {partition} '
+                                           f'is out of bounds [{earliest_offset}, {oldest_offset})')
+
+    return True
+
+
+def fetch_incidents(kafka: KafkaCommunicator, demisto_params: dict) -> None:
+    """Fetch incidents as kafka messages from a specific topic.
+
+    Args:
+        kafka: initialized KafkaCommunicator object to preform actions with.
+        demisto_params: The demisto parameters.
+    """
+    topic = demisto_params.get('topic', '')
+    partitions = handle_empty(argToList(demisto_params.get('partition', '')), -1)
+    offset = handle_empty(demisto_params.get('first_fetch', 'earliest'), 'earliest')
+    message_max_bytes = int(handle_empty(demisto_params.get("max_bytes_per_message", 1048576), 1048576))
+    max_messages = int(handle_empty(demisto_params.get('max_fetch', 50), 50))
+    last_fetched_offsets = demisto.getLastRun().get('last_fetched_offsets', {})
+    last_topic = demisto.getLastRun().get('last_topic', '')
+    demisto.debug(f"Starting fetch incidents with last_fetched_offsets: {last_fetched_offsets}")
```

Comment:
lets log here all the params that are related to the fetch (max messages, topic, etc..)